addon/components/file-picker.js

```diff
@@ -96,10 +102,6 @@ export default Component.extend({
           });
       }
     }
-
-    if (!this.get('hideFileInput')) {
-      this.$('.file-picker__input').val(null);
-    }
```

Comment:
It was unclear to me why the visibility of the input control was taken into account here. It seems that we'll want to reset the input's value regardless of whether the input control is hidden or not.

This was also clearing the value _after_ the selection. So if the control was visible the selected file path would not be shown.

Reply:
@jrowlingson I guess that was a way to fix the webkit caching but you are totally right that it shouldn't depend on the visibility.